sc-memory/sc-memory/src/scs/scs_parser.cpp

```diff
@@ -413,44 +456,109 @@ ElementHandle Parser::ProcessIdentifier(std::string const & name)
 }
 
-ElementHandle Parser::ProcessIdentifierLevel1(std::string const & type, std::string const & name)
+ElementHandle Parser::ProcessIdentifierLevel1(std::string const & scsType, std::string const & name)
 {
-  ScType elType = scs::TypeResolver::GetKeynodeType(type);
-  elType |= scs::TypeResolver::IsConst(name) ? ScType::Const : ScType::Var;
-
-  return AppendElement(name, elType);
+  ScType type = scs::TypeResolver::GetElementType(scsType);
+  type |= scs::TypeResolver::IsConst(name) ? ScType::Const : ScType::Var;
+  return AppendElement(name, type);
 }
 
-void Parser::ProcessTriple(ElementHandle const & source, ElementHandle const & connector, ElementHandle const & target)
+void Parser::ProcessTriple(
+    ElementHandle const & sourceHandle,
+    ElementHandle const & connectorHandle,
+    ElementHandle const & targetHandle)
 {
-  ParsedElement const & connectorEl = GetParsedElement(connector);
+  ParsedElement & connector = GetParsedElementRef(connectorHandle);
 
-  auto AddConnector = [this, &connectorEl](ElementHandle const & src, ElementHandle const & e, ElementHandle const & trg) {
-    ParsedElement const & srcEl = GetParsedElement(src);
-    std::string const & idtf = srcEl.GetIdtf();
-    if (connectorEl.GetType() == ScType::ConstPermPosArc && scs::TypeResolver::IsKeynodeType(idtf))
-    {
-      ParsedElement & targetEl = GetParsedElementRef(trg);
-      ScType const newType = targetEl.m_type | scs::TypeResolver::GetKeynodeType(idtf);
+  auto AddConnector =
+      [this, &connector](
+          ElementHandle const & sourceHdl, ElementHandle const & connectorHdl, ElementHandle const & targetHdl)
+  {
+    ParsedElement const & source = GetParsedElement(sourceHdl);
+    ParsedElement & target = GetParsedElementRef(targetHdl);
 
-      if (targetEl.m_type.CanExtendTo(newType))
-        targetEl.m_type = newType;
-      else
-        SC_THROW_EXCEPTION(utils::ExceptionParseError, "Can't merge types for element " + targetEl.GetIdtf());
+    if (source.IsElementType() && (connector.GetType() == ScType::ConstPermPosArc))
+    {
+      std::string const & sourceIdtf = source.GetIdtf();
+      ScType const sourceType = scs::TypeResolver::GetElementType(sourceIdtf);
+      ScType const targetType = target.m_type;
+      ScType const newTargetType = targetType | sourceType;
+
+      if (target.IsElementType() && !newTargetType.CanExtendTo(ScType::ConstNodeClass))
+        SC_THROW_EXCEPTION(
+            utils::ExceptionParseError,
+            "Can't extend type `" << std::string(targetType) << "` using type `" << std::string(sourceType)
+                                  << "` for specified sc-element, because sc-element `" << target.GetIdtf()
+                                  << "` is sc-element denoting type of sc-elements.");
+      else if (ScType(targetType.BitAnd(~(ScType::Const | ScType::Var))).CanExtendTo(sourceType))
+        target.m_type = newTargetType;
+      else if (!sourceType.CanExtendTo(targetType))
```

Comment:
I've added VarPermPosArc to the set of ConstPermPosArcs and got no error
![image](https://github.com/user-attachments/assets/82012c21-d07c-4db2-a930-296d2fe38ca4)

Reply:
VarPermPosArc can't be extended ConstPermPosArc

Reply:
I know it can't, but sc-builder is saying 'ok' for that triple